loadtest/control/actions.go

```diff
@@ -492,6 +492,21 @@ func SearchPosts(u user.User) UserActionResponse {
 	return UserActionResponse{Info: fmt.Sprintf("found %d posts", len(list.Posts))}
 }
 
+// GetPinnedPosts fetches the pinned posts in a channel that user is a member of.
+func GetPinnedPosts(u user.User) UserActionResponse {
+	channel, err := u.Store().CurrentChannel()
+	if err != nil {
+		return UserActionResponse{Err: NewUserError(err)}
+	}
```

Comment:
`CurrentChannel` can return `nil` without that being an error. So we should always check that before going any further.

Reply:
@streamer45 What about returning an error if we can't find the current channel? I would say it is an error if I can't get channel from `CurrentChannel()`. Interestingly, it never returns an error at the moment.

Reply:
We could do that but then every time we want to know if it's set we need to check for a specific error instead of checking for nil. This is how I used it:
 
https://github.com/mattermost/mattermost-load-test-ng/blob/9a452a99136830ab77326a6faa1c8023a06bdea8/loadtest/control/actions.go#L550-L554

As of why it's not failing I have no idea. I believe I removed every instance of `SetCurrentChannel` from my previous PR.

Reply:
We can consider not being able to getting `CurrentChannel()` is an error, if there is a channel, then we can consider no error. We can keep the usage like you have already.

Reply:
We can but I think there might be instances where getting no channel is just because we haven't set it yet and we might want to set it in response to that. 
That said, the logic is very similar whether we return an error or not. I am okay with both approaches.

Reply:
If you are okay, I will be in the side of returning an error when there is no channel :)